### DISCUSSION TITLE
How to implement Single Signout for SAML flow.

### DISCUSSION THREAD

**Discussion by yogeshuchate** [ory/hydra#2972] [Q&A] [answered]
Hi Ory Hydra Team,
We are using Ory Hydra as OIDC. We have legacy app which take cares of all the SAML flow like sign-in and logout.
We just added a redirection from Ory Hydra OIDC to our legacy application for SAML flow. For logout OIDC api are working fine but we want to clear the legacy application session. Can you please let me know if there is any way to do Single Sing Out for Ory Hydra.

Thanks,
Yogesh

**Comment by vinckr** [comment, marked as answer]
Hello Yogesh,
I am not the expert but AFAIK this is really hard to implement with OAuth2.0/OIDC. 
We have [Ory Kratos](https://github.com/ory/kratos) and there is an [ongoing effort](https://github.com/ory/kratos/pull/2148) to implement SAML. I think this would be a good solution and supports "Instant Logout".
There is also a way to solve this with Ory Hydra alone, by keeping a check on the "logged out"/invalidated sessions, but I am not sure how to do it exactly.

**Reply by yogeshuchate** [reply]
Thanks Vincent for reply.
You mean we cant clear the SAML session using Ory Hydra & we can use Ory Kratos... !
We already invalidating the Ory Hydra session by calling **/oauth2/auth/requests/logout** .
Only concern is legacy app and SAML session . ..!

Thanks,
Yogesh
